File: src/branch.rs
pub enum BranchType {
    Base,
    Target,
}

pub struct Branch {
    pub name: String,
    pub branch_type: BranchType,
}

impl Branch {
    pub fn app_file(&self) -> &'static str {
        match self.branch_type {
            BranchType::Base => "apps_base_branch.yaml",
            BranchType::Target => "apps_target_branch.yaml",
        }
    }

    pub fn folder_name(&self) -> &str {
        match self.branch_type {
            BranchType::Base => "base-branch",
            BranchType::Target => "target-branch",
        }
    }
}

impl std::fmt::Display for BranchType {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        match self {
            BranchType::Base => write!(f, "base"),
            BranchType::Target => write!(f, "target"),
        }
    }
}

File: pkg/types/branch.go
package types

import "fmt"

// Branch represents a git branch and its local folder
type Branch struct {
	Name       string
	folderName string
	branchType BranchType
}

type BranchType string

const (
	Base   BranchType = "base"
	Target BranchType = "target"
)

// NewBranch creates a new Branch instance
func NewBranch(name string, branchType BranchType) *Branch {
	return &Branch{
		Name:       name,
		folderName: fmt.Sprintf("%s-branch", branchType),
		branchType: branchType,
	}
}

// FolderName returns the folder name for the branch
func (b *Branch) FolderName() string {
	return b.folderName
}

func (b *Branch) Type() BranchType {
	return b.branchType
}
